subprojects/docs/src/docs/dsl/org.gradle.plugins.ide.idea.model.IdeaModule.xml

```diff
@@ -17,7 +17,7 @@
             <tr>
                 <td>sourceDirs</td>
                 <td><literal>[]</literal></td>
-                <td>source dirs from <literal>project.sourceSets.main.allSource</literal></td>
+                <td>source dirs from <literal>project.sourceSets.main.allJava</literal></td>
```

Comment:
❌ Maybe I'm missing something, but this looks like a breaking change. IntelliJ currently calls `getSourceDirs` to determine all source directories and they don't yet know about the new `getResourceDirs` method. That means Gradle 4.7 projects would fail to import into IntelliJ.

Reply:
@fanick1 I'm sorry that I missed this, but @oehme is correct. This PR should be modified as follows:
- `sourceDirs` and `testSourceDirs` should have the original content (`allSource`).
- The documentation for the getter methods should explicitly say, that `allSource` is returned only for compatibility reasons and `allJava` will be returned in Gradle 5.0.

Reply:
I'look into this today. Should I also cover this future change in a cross version test annotated with target api ≥ 5.0?

Reply:
That's a great idea, because it'll remind us to change it when 5.0 comes around.

Reply:
fixed